paddle/fluid/framework/ir/mkldnn/compute_propagate_scales_mkldnn_pass.cc

```diff
@@ -0,0 +1,426 @@
+  auto* data = tensor->data<float>();
+  const auto dims = tensor->dims();
+  PADDLE_ENFORCE_EQ(dims.size(), 2,
+                    "The input tensor's rank is required to be 2.");
+
+  const int rows = dims.at(0);
+  const int columns = dims.at(1);
+  std::vector<float> scales;
+  if (axis == 0) {
+    for (int i = 0; i < columns; i++) {
+      float max_value = FLT_MIN;
+      for (int j = 0; j < rows; j++) {
+        max_value = std::max(max_value, std::abs(data[i + j * columns]));
+      }
+      max_value = 1.0 / max_value;
+      if (std::isinf(max_value) || std::isnan(max_value)) {
+        max_value = 0.0;
+      }
+      scales.push_back(max_value);
+    }
+  } else {
+    for (int i = 0; i < rows; i++) {
+      float max_value = FLT_MIN;
+      for (int j = i * columns; j < (i + 1) * columns; j++) {
+        max_value = std::max(max_value, std::abs(data[j]));
+      }
+      max_value = 1.0 / max_value;
+      if (std::isinf(max_value) || std::isnan(max_value)) {
+        max_value = 0.0;
+      }
+      scales.push_back(max_value);
+    }
+  }
+  return scales;
+}
+
+void ComputePropagateScalesMkldnnPass::ComputeVarScales(
+    ir::Graph* graph, Scope* scope, const std::unordered_set<std::string>& ops,
+    const std::string& weight_name, const int axis,
+    StringPairMap* var_quant_scales) const {
+  for (auto* op_node :
+       ir::TopologyVarientSort(*graph, static_cast<ir::SortKind>(0))) {
+    if (!op_node->IsOp()) continue;
+
+    auto* op_desc = op_node->Op();
+    if (ops.count(op_desc->Type())) {
+      auto var_name = op_desc->Input(weight_name)[0];
+      auto* var = scope->FindVar(var_name);
+      PADDLE_ENFORCE_NOT_NULL(
+          var, "The input persistable var of %s op is not found.",
+          op_desc->Type());
+      auto* weight_tensor = var->GetMutable<LoDTensor>();
+      const auto dims = weight_tensor->dims();
+      int volume = 1;
+      for (int i = 1; i < dims.size(); i++) {
+        volume *= dims[i];
+      }
+
+      Tensor tmp_tensor;
+      std::vector<int64_t> reshape_dims = {dims[0], volume};
+      tmp_tensor.Resize(phi::make_ddim(reshape_dims));
+      auto* weight_data = weight_tensor->data<float>();
+      auto* tmp_data = tmp_tensor.mutable_data<float>(platform::CPUPlace());
+      for (int i = 0; i < weight_tensor->numel(); i++) {
+        tmp_data[i] = std::abs(weight_data[i]);
+      }
+
+      auto scales_v = GetScales(&tmp_tensor, axis);
+      Tensor tensor;
+      GetTensorFromVector(scales_v, &tensor);
+      auto pair = std::make_pair(false, tensor);
+      var_quant_scales->insert(std::make_pair(var_name, pair));
+    }
+  }
+}
+
+void ComputePropagateScalesMkldnnPass::ComputeSingleGruWeightScales(
+    Scope* scope, const std::string& wx_var_name,
+    const std::string& wh_var_name, Tensor* tensor) const {
+  auto* wx_var = scope->FindVar(wx_var_name);
+  PADDLE_ENFORCE_NOT_NULL(wx_var, "The input persistable var %s is not found.",
```

Comment:
same above, please refer to the result of approve CI check and modify it in turn